Userland/userdel.cpp

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2019-2020, Fei Wu <f.eiwu@yahoo.com>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <AK/String.h>
+#include <AK/StringBuilder.h>
+#include <LibCore/ArgsParser.h>
+#include <ctype.h>
+#include <dirent.h>
+#include <errno.h>
+#include <pwd.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+static int delete_user(const char* username, bool remove_home);
+
+int main(int argc, char** argv)
+{
+    const char* username = nullptr;
+    bool remove_home = false;
+
+    Core::ArgsParser args_parser;
+    args_parser.add_option(remove_home, "Remove home directory", "remove", 'r');
+    args_parser.add_positional_argument(username, "Login user identity (username)", "login");
+    args_parser.parse(argc, argv);
+
+    return delete_user(username, remove_home);
+}
+
+int delete_user(const char* username, bool remove_home)
+{
+    const char* temp_filename = "passwd.temp";
+    FILE* temp_file = fopen(temp_filename, "w");
+    if (!temp_file) {
+        perror("failed to create temporary file");
+        return 1;
+    }
```

Comment:
Use `mkstemp()` instead. For one thing, it'll use `O_EXCL, S_IRUSR | S_IWUSR` properly.